sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/state/ViewportState.kt

```diff
@@ -0,0 +1,25 @@
+package com.mapbox.maps.plugin.viewport.experimental.state
+
+import com.mapbox.maps.plugin.animation.Cancelable
+import com.mapbox.maps.plugin.viewport.ViewportPlugin
+
+/**
+ * Set of possible [ViewportPlugin] states.
+ */
+interface ViewportState {
+  /**
+   * Observer the new camera options produced by the [ViewportState], it can be used to get the
+   * latest state [CameraOptions] for [ViewportTransition].
+   *
+   * @param viewportDataSourceObserver observer that observe new viewport data.
+   * @return a handle that cancels current observation.
+   */
+  fun observeDataSource(viewportDataSourceObserver: ViewportDataSourceObserver): Cancelable
+
+  /**
+   * Start updating the camera for the current [ViewportState].
+   *
+   * @return a handle that cancels the camera updates.
+   */
+  fun startUpdatingCamera(): Cancelable
```

Comment:
We use a cancelable here so that the state can handle the proper cleanup, unsubscribe from location updates etc.
I'm also fine with moving it to `ViewportState`, cc @macdrevx

Reply:
@LukasPaczos that's a good point. On iOS, I even have the following right now in the FollowingViewportState implementation:

```
    // viewport should only call this method when the state is not already updating the camera
    public func startUpdatingCamera() -> Cancelable {
        assert(!isUpdatingCamera)
        isUpdatingCamera = true
```

That's not ideal.

@pengdev what do you think about changing this interface to have 2 methods:

```
startUpdatingCamera()
stopUpdatingCamera()
```

Within Viewport, we can still wrap this with Cancelable if it makes things easier, but I agree that this interface would make it more clear that states can't have parallel updates running.

Reply:
👍 sounds good to me, if we add the stopUpdatingCamera, maybe we should remove the cancelable returned?

Reply:
Yep